packages/grafana-ui/src/components/DataSourceSettings/CustomHeadersSettings.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { PureComponent } from 'react';
+import { Button } from '../Button/Button';
+import { FormField } from '../FormField/FormField';
+import { SecretFormField } from '../SecretFormFied/SecretFormField';
+import { DataSourceSettings } from '@grafana/data';
+
+export interface CustomHeader {
+  name: string;
+  value: string;
+  configured: boolean;
+}
+
+export type CustomHeaders = CustomHeader[];
+
+export interface Props {
+  dataSourceConfig: DataSourceSettings<any, any>;
+  onChange: (config: DataSourceSettings) => void;
+}
+
+export interface State {
+  headers: CustomHeaders;
+}
+
+export class CustomHeadersSettings extends PureComponent<Props, State> {
+  state: State = {
+    headers: [],
+  };
+
+  constructor(props: Props) {
+    super(props);
+  }
+
+  componentDidMount(): void {
+    this.updateHeaders();
+  }
+
+  updateSettings(headers: CustomHeaders) {
+    const { jsonData, secureJsonData } = this.props.dataSourceConfig;
```

Comment:
`secureJsonData` can be undefined, hence I am getting this error when adding first header:

![image](https://user-images.githubusercontent.com/2376619/68937067-0ba6e100-079c-11ea-8b24-d6f7715b3a48.png)